sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/common/attribute_value.h"
+
+#include <memory>
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+class Sampler
+{
+public:
+  /**
+   * Initialize a new tracer.
+   * @param processor The span processor for this tracer. This must not be a
+   * nullptr.
+   */
+  virtual Decision shouldSample();
+
+  /**
+   * Returns the description of this {@code Sampler}. This may be displayed on debug pages or in the
+   * logs.
+   *
+   * <p>Example: "ProbabilitySampler{0.000100}"
+   *
+   * @return the description of this {@code Sampler}.
+   */
+  virtual std::string getDescription();
```

Comment:
Make method names start with an uppercase letter and `noexcept`. Make virtual methods pure virtual. So this should read:
```c++
virtual std::string GetDescription() const nexcept = 0;
```